### PR TITLE
http->https redirect

### LINKED ISSUE #177: HTTP to HTTPS redirect
### Enhancement Proposal

After tls is enabled in traefik, a rule could be created for HTTP to HTTPS redirects. I see no reason not to do that.

An example of how to implement that can be found [here](https://jensknipper.de/blog/traefik-http-to-https-redirect/).

### PR DESCRIPTION
**Pull request by PietroPasotti**
Configures traefik to redirect by default http to https if tls is enabled.

Replaces #246 
Fixes #177

### CODE REVIEW

File: src/traefik.py [hunk 1]
```diff
@@ -199,14 +199,20 @@ def _update_static_configuration(self):
         tcp_entrypoints = self._tcp_entrypoints
         logger.debug(f"Statically configuring traefik with tcp entrypoints: {tcp_entrypoints}.")
 
+        web_config = {
+            "address": f":{self.port}",
+            # enable by default http -> https redirect
+            "http": {"redirections": {"entryPoint": {"to": "websecure", "scheme": "https"}}},
```
Comment (mmkay) [inline src/traefik.py:205]: Might be missing something, but should we add a redirection scheme only if there is a certificate? I could imagine a tricky redirect look for a HTTP-only setup if traefik doesn't take into account whether certificate exists.

Comment (PietroPasotti) [reply src/traefik.py:205]: that might explain why the itests are failing though :D

**Review by mmkay — APPROVED** [review]

**Comment by PietroPasotti** [comment]
this entails that traefik won't route http requests anymore without upgrading them to https.
@simskij we seem to recall that this change was rejected in the past on the grounds that some applications might actually prefer to be routed over http even if https is available. Can you confirm?

**Comment by simskij** [comment]
> this entails that traefik won't route http requests anymore without upgrading them to https. @simskij we seem to recall that this change was rejected in the past on the grounds that some applications might actually prefer to be routed over http even if https is available. Can you confirm?

I think that was expressed in the context of end-to-end TLS, where we can't necessarily guarantee that everything being routed by traefik supports TLS for the communication between traefik itself and the workload. As for the ingress, upgrading to TLS when configured is in my eyes a reasonable decision.

**Pull request merged**